zproject/computed_settings.py

```diff
@@ -699,6 +698,16 @@ def zulip_path(path: str) -> str:
 SCIM_LOG_PATH = zulip_path("/var/log/zulip/scim.log")
 
 ZULIP_WORKER_TEST_FILE = zulip_path("/var/log/zulip/zulip-worker-test-file")
+
+ANALYTICS_LOCK_FILE = zulip_path("/home/zulip/deployments/analytics-lock.lock")
+ARCHIVE_MESSAGES_LOCK_FILE = zulip_path("/home/zulip/deployments/archive-messages.lock")
+DELETE_ATTACHMENTS_LOCK_FILE = zulip_path("/home/zulip/deployments/delete-attachments.lock")
+DOWNGRADE_REALMS_LOCK_FILE = zulip_path("/home/zulip/deployments/downgrade-realms.lock")
+ENQUEUE_DIGEST_EMAILS_LOCK_FILE = zulip_path("/home/zulip/deployments/send-digest-emails.lock")
+FIRST_VISIBLE_MESSAGE_LOCK_FILE = zulip_path("/home/zulip/deployments/first-visible-mesage.lock")
+INVOICE_PLANS_LOCK_FILE = zulip_path("/home/zulip/deployments/invoice-plans.lock")
+PROMOTE_USERS_LOCK_FILE = zulip_path("/home/zulip/deployments/promote-users.lock")
+SOFT_DEACTIVATE_LOCK_FILE = zulip_path("/home/zulip/deployments/soft-deactivate.lock")
 ZULIP_UPDATE_ANNOUNCEMENTS_LOCK_FILE = zulip_path(
     "/home/zulip/deployments/zulip_update_announcements.lock"
 )
```

Comment:
I think you mentioned the thought that these should be a directory; I definitely agree it'd be nice to have a `FILE_LOCK_DIRECTORY = zulip_path("/home/zulip/deployments/lockfiles")` sort of structure, rather than adding a new setting for each one.

Reply:
Yeah, that will avoid both pollution of variables as well as pollution of `/home/zulip/deployments`